### PR TITLE
refactor(agent): contract direct Convex transports

### PR DESCRIPTION
**Pull request by nabilfatih**
## Summary

- make Convex HTTP Actions the only public REST and MCP protocol implementation
- keep the API and MCP Vercel apps only as hostname, secret, security-header, and rewrite boundaries
- expose one public REST version at `/v1` and one MCP transport at `/mcp`

## Why

Nakafa had duplicate protocol implementations in Next.js and Convex. Those Next routes, the MCP registry, and bridge libraries could drift from the authenticated Convex contracts. PR #373 introduced public API V2 only as a migration expansion surface. The operator accepts predecessor breakage and requires contraction now, so V2 and legacy advertising are deleted instead of becoming permanent debt.

## Scope

- delete the Next public MCP server, tool registry, REST handlers, bridge libraries, obsolete tests, unused dependencies, and redundant private Convex MCP health route
- preserve the separate private signed-content runtime
- rewrite only `/openapi.json` and `/v1/*` through the protected API bridge, and `/mcp` through the protected MCP bridge
- derive the forwarded host from the trusted request URL so Next 16.3.2 external rewrites preserve Convex `/internal/*` paths without forwarding an untrusted Host header
- delete direct Convex REST predecessor access, public `/v2`, its OpenAPI contract, schemas, matcher, examples, and stale tool advertisements
- reject predecessor MCP protocol requests while preserving the surviving Convex tool contract
- explicitly reject root `/quran/*`; PR #390 owns Quran behavior only within canonical `/v1/quran/:surah`
- retain one bounded internal mutation only to delete the exact production observer row after protected deployment

## Exact-head verification

Head: `0e930ab34a627bff6c3c20b4bbaef55d0a577fc3`, based directly on protected main `e4447e71d52d5facdf37c2776b5835ca8d3bc394`.

The 11 transport patches remained patch-identical across the rebase. Post-rebase proof is green: 48 focused backend transport/OpenAPI tests; the complete API suite with 57 tests at 100 percent coverage; the complete MCP suite with 4 tests at 100 percent coverage; backend, API, and MCP typechecks; lint; Effect 4.0.0-rc.110 source parity; diff checks; explicit `/v2` and root `/quran` rejection; and touched handwritten files at or below 500 lines. Earlier full proof also covered 353 backend files and 1,510 tests, production builds, boundaries, security audit, and isolated Convex API/MCP acceptance. Exact-head CI run `33071240715` is green: Quality, signed Production build/start, browser acceptance, AFDocs, final runtime verification, cleanup, and Required all passed. Doctor run `33071240706` is green. Fresh exact-head Codex review comment `5439349284` reviewed `0e930ab34a` and found no major issues. All three earlier findings were independently classified with evidence and remain resolved.

## Rollout state

Protected rebase merge completed at `e5155c5a7905d66ee2688f1ec11999b5a8259f06`; its tree exactly equals reviewed head `0e930ab34a627bff6c3c20b4bbaef55d0a577fc3`. No Vercel Preview or production mutation was used. Exact-main CI and Git-integration production deployments are now running. Live V1 and MCP acceptance must pass before invoking the exact retirement mutation for `mcp-direct-1f955564`. An immediate cleanup PR then deletes that mutation, observer schema and table, generated binding, and tests after the row is removed and production is verified empty.

## Risks

Old direct-origin REST and predecessor MCP clients intentionally stop working. Quran domain changes remain outside this transport PR. The final public contract has no API V2, root Quran alias, or `_v2` MCP advertisement.

### CODE REVIEW

File: apps/mcp/lib/mcp/tools/register.ts [hunk 1]
```diff
@@ -1,139 +0,0 @@
...
-    outputSchema: NakafaGetQuranReferenceToolOutputSchema,
-    run: getNakafaQuranReferenceToolResult,
-    title: "Get Nakafa Quran Reference",
-  },
-  {
-    annotations: NAKAFA_READ_ONLY_TOOL_ANNOTATIONS,
-    description:
-      "Preferred Quran tool. Return semantic translation notes, signed Arabic and translation sources, and explicit embedded or official link-only tafsir access.",
-    inputSchema: NakafaGetQuranReferenceV2ToolInputSchema,
-    name: "nakafa_get_quran_reference_v2",
```
[the remark this answers is not in the document]
Comment (nabilfatih) [reply apps/mcp/lib/mcp/tools/register.ts:96]: Valid at the reviewed head. The direct Convex MCP server registered only `nakafa_get_quran_reference`, while the informational root and both taxonomy projections still advertised `_v2`. Exact head `9dc1885bbd` fixes the contract by deleting the public `/v2` route, OpenAPI V2 path/schema/example, edge matcher, and every `_v2` tool advertisement. It does not add another compatibility tool. Focused API, MCP, OpenAPI, taxonomy, and Convex tests are green, and the surviving MCP tool list remains exact.

File: packages/backend/convex/routes/agent/mcp/route.ts [hunk 2]
```diff
@@ -118,7 +106,7 @@ export function registerAgentMcpRoutes(app: AgentApp) {
       () =>
         runtime.success.server.createNakafaMcpServer(context.env, requestId),
       {
-        legacy: "stateless",
+        legacy: "reject",
```
[the remark this answers is not in the document]
Comment (nabilfatih) [reply packages/backend/convex/routes/agent/mcp/route.ts:87]: The mechanism is correctly identified, but the requested wait is no longer the operator contract. The repository owner explicitly authorized immediate predecessor breakage and removal of backward compatibility, including rejection before a 24-hour quiet window. Exact head `9dc1885bbd` therefore intentionally retains `legacy: "reject"`; focused MCP tests prove missing-version and old-version requests fail with the documented 400 responses. This is an explicit contract decision, not an unobserved accidental regression.

File: packages/backend/convex/http.ts [hunk 3]
```diff
@@ -50,10 +49,7 @@ app.on(["POST", "GET"], "/api/auth/*", (c) => {
   return auth.handler(c.req.raw);
 });
 
-// Preserve the deployed predecessor until the protected Vercel bridge switches.
-app.route("/v1", v1);
-
-// Register the protected successor and its machine-readable contract.
+// Register the protected public protocols and their machine-readable contract.
 registerAgentApiRoutes(app);
 registerAgentMcpRoutes(app);
```
[the remark this answers is not in the document]
Comment (nabilfatih) [reply packages/backend/convex/http.ts:54]: The direct Convex-site `/v1` break is intentional and explicitly authorized. The supported public contract is `https://api.nakafa.com/v1` through the protected Vercel edge bridge, and the operator accepted breakage for unused direct-origin callers in order to remove all legacy surfaces now. Exact head `9dc1885bbd` proves the old origin mount returns 404 while the protected `/v1` rewrite and API contract remain tested. Retaining a new observation seam would directly contradict that requested contraction.

**Comment by nabilfatih** [comment]
@codex review

**Comment by nabilfatih** [comment]
Production contraction completed at exact main `e5155c5a7905d66ee2688f1ec11999b5a8259f06`.

- merge tree `e5d9f3647a86b0a20efabb9889e5d7e41fc927da` exactly matches reviewed head `0e930ab34a627bff6c3c20b4bbaef55d0a577fc3`
- merge-level CI `33073477724` passed Quality, Production, Browser, AFDocs, final runtime verification, cleanup, and Required; Doctor `33073477819` passed
- exact-main Vercel deployments are READY for `www`, `api`, `mcp`, and `cas`
- live OpenAPI has only seven `/v1` paths; `/v2` and root `/quran` return 404; canonical search and Quran reads return 200
- direct and recommended MCP endpoints report exact main and the four canonical tools; predecessor protocol `2025-11-25` returns the expected 400
- exact-deployment API, MCP, and WWW error/fatal logs are empty
- Convex production exposes only `routes/agent/mcp/legacy.js:retire` from the temporary lifecycle

The exact operator mutation retired `mcp-direct-1f955564` with server receipt: `deleted=1`, `invocationCount=141`, `armedAt=1787799652585`, `lastInvokedAt=1787831151637`, `quietSince=1787831151637`, `retiredAt=1787836328369`, deployment `dapper-antelope-269`. A production data read now reports no documents in `agentMcpLegacyReads`.

The immediate cleanup PR will delete the remaining mutation, schema, table binding, generated registration, and tests.
